### ISSUE TITLE
`reqwest::Response` is not `Send` in WASM target

### ISSUE THREAD

**Issue by zhangzhonglai** [marc2332/dioxus-query#6]
When I use `reqwest::get` in boxed future, it throws a compile error. Because type `request::Response` is not `Send` in wasm target.

```
Error: 🚫 Building project failed: error: future cannot be sent between threads safely
   --> frontend/src/page/api/index.rs:195:5
    |
195 | /     Box::pin({
...   |
202 | |     })
    | |______^ future created by async block is not `Send`
    |
    = help: within `[async block@frontend/src/page/api/index.rs:196:9: 201:10]`, the trait `std::marker::Send` is not implemented for `*mut u8`
note: future is not `Send` as this value is used across an await
   --> frontend/src/page/api/index.rs:209:10
    |
206 |       let res = reqwest::get(format!("{}", SERVER_URL))
    |  _______________-
207 | |         .await?
    | |_______________- has type `Response` which is not `Send`
209 |           .await?;
    |            ^^^^^ - the value is later dropped here
    |            |
    |            await occurs here, with the value maybe used later
    = note: required for the cast from `Pin<std::boxed::Box<[async block@frontend/src/page/api/index.rs:196:9: 201:10]>>` to `Pin<std::boxed::Box<dyn futures_util::Future<Output = QueryResult<std::vec::Vec<entity::api::Model>, ()>> + std::marker::Send>>`
```

Any suggestion on how we can avoid this?

**Comment by marc2332** [comment]
Can you share me the code from the query function?

**Comment by marc2332** [comment]
Can you try with https://github.com/marc2332/dioxus-query/pull/7 ?

Cargo.toml:
```toml
dioxus-query = { git = "https://github.com/marc2332/dioxus-query", branch = "feat/relax-trait-bounds" }
```

**Comment by zhangzhonglai** [comment]
#7 has no effect. I think the problem is `BoxFuture` required a future with `Send`, but `reqwest::Response` in wasm platform is not `Send`. Part of the code snippet:

``` rust

fn fetch_api_list(keys: &[QueryKeys]) -> BoxFuture<QueryResult<Vec<Api>, ()>> {
    Box::pin({
        async move {
            get_api_list().await.map_err(|e| {
                error!("Error occurred: {e}");
                ()
            }).into()
        }
    })
}

async fn get_api_list() -> Result<Vec<Api>, Error> {
    let res = reqwest::get(format!("{}/api/api", SERVER_URL))
        .await?
        .json::<JsonResponse<Vec<Api>>>()
        .await?;
    if res.success {
        Ok(res.data)
    } else {
        Ok(vec![])
    }
}
```

**Comment by marc2332** [comment]
I'll see what I can do

**Comment by marc2332** [comment]
heyy! I think I got it, try with https://github.com/marc2332/dioxus-query/pull/8 ! I have even simplified the API

**Comment by zhangzhonglai** [comment]
Wow, it works, and much simpler and clearer than before. Nice work!

**Comment by marc2332** [comment]
![](https://media1.giphy.com/media/l2Je1Jyuhxd3caP1C/giphy.gif?cid=ecf05e47jjfdi3ly6khfru393s7ccxn59bkq2xus7aew71s7&ep=v1_gifs_search&rid=giphy.gif&ct=g)

**Comment by marc2332** [comment]
I'll merge it and make a release asap!

**Comment by marc2332** [comment]
hi @zhangzhonglai , I appreciate that you reported the issue, thank you!

 I just released the new version [`v0.3.2`](https://crates.io/crates/dioxus-query) ✨
